src/app/components/query-explainer/query-explainer.component.ts

```diff
@@ -0,0 +1,129 @@
+                                );
+                            });
+                    }),
+                    switchMap((response: QueryExplainerResponse) => {
+                        return this.queryExplainerService.verifyQuery(response);
+                    }),
+                    catchError((e: HttpErrorResponse) => {
+                        return of(e.error as VerifyQueryResponse);
+                    }),
+                    tap((data) => {
+                        this.verifyResponse = data;
+                    }),
+                    switchMap(() => forkJoin(blockHashObservables$)),
+                    tap((blockHashSystemTimes) => {
+                        this.blockHashSystemTimes = blockHashSystemTimes;
+                    }),
+                    takeUntilDestroyed(this.destroyRef),
+                )
+                .subscribe(() => this.cdr.detectChanges());
+        }
+    }
+
+    public copyToClipboard(event: MouseEvent, text: string): void {
+        this.clipboard.copy(text);
+        if (event.currentTarget !== null) {
+            const currentElement: HTMLButtonElement = event.currentTarget as HTMLButtonElement;
+            const currentElementChildren: HTMLCollectionOf<HTMLElement> =
+                currentElement.children as HTMLCollectionOf<HTMLElement>;
+            setTimeout(() => {
+                currentElementChildren[0].style.display = "inline-block";
+                currentElementChildren[1].style.display = "none";
+                currentElement.classList.remove("clipboard-btn--success");
+            }, AppValues.LONG_DELAY_MS);
+            currentElementChildren[0].style.display = "none";
+            currentElementChildren[1].style.display = "inline-block";
+            currentElement.classList.add("clipboard-btn--success");
+        }
+    }
+
+    private extractSqlQueryFromRoute(): MaybeNull<string> {
+        const sqlQueryFromRoute: MaybeUndefined<string> = this.activatedRoute.snapshot.queryParams[
+            ProjectLinks.URL_QUERY_PARAM_SQL_QUERY
+        ] as MaybeUndefined<string>;
+        return sqlQueryFromRoute ?? "";
+    }
+
+    public routeToDataset(alias: string): string {
+        return alias.includes("/") ? alias : `kamu/${alias}`;
+    }
+
+    public get schemaFields(): DataSchemaField[] {
+        return extractSchemaFieldsFromData(this.tableSource[0]);
+    }
+
+    public inputParamsHelper(option: string): string {
+        switch (option) {
+            case "SqlDataFusion":
+                return "Sql Data Fusion";
+            case "JsonAoA":
+                return "Json AoA";
+            case "ArrowJson":
+                return "Arrow Json";
+            default:
+                return "Unknown options";
+        }
+    }
+
+    private columnNames(): string[] {
+        return this.sqlQueryExplainer?.output?.schema.fields.map((item) => item.name) as string[];
+    }
```

Comment:
Make it accept a parameter that is already checked against null/undefined.
I.e. `this.sqlQueryExplainer.output.` could come from HTML template within checked divs.